libmamba/src/core/util_string.cpp

```diff
@@ -15,6 +15,27 @@
 
 namespace mamba
 {
+    /********************************************************
+     *  Implementation of Channels use case util function   *
+     *******************************************************/
+    std::string get_common_substr(std::string_view str1, std::string_view str2)
```

Comment:
Three more nitpicking suggestions:
- Looking more into the function, maybe a more explicit name could be `get_common_parts`?
- `'/'` could then be a `delim` parameter
- Possibly, the function could return a `std::string_view` as well, avoiding the need to create a string when the user does not need it.